### ISSUE TITLE
System.NotSupportedException is thrown by UsersManager.GetCurrentUserInformationAsync if the SDK is used in a single-file bundle

### ISSUE THREAD

**Issue by enotesynctest2** [box/box-windows-sdk-v2#905]
- &#9745; I have checked that the [SDK documentation][sdk-docs] doesn't solve my issue.
- &#9745; I have checked that the [API documentation][api-docs] doesn't solve my issue.
- &#9745; I have searched the [Box Developer Forums][dev-forums] and my issue isn't already reported (or if it has been reported, I have attached a link to it, for reference).
- &#9745; I have searched [Issues in this repo][github-repo] and my issue isn't already reported.

### Description of the Issue
I'm using the SDK in my .NET 6 app published as a [single-file bundle](https://learn.microsoft.com/en-us/dotnet/core/deploying/single-file/overview?tabs=cli). 
When I'm doing this:

`BoxUser currentUser = await client.UsersManager.GetCurrentUserInformationAsync();`

the System.NotSupportedException ("CodeBase is not supported on assemblies loaded from a single-file bundle") is thrown. 

It is thrown in the Box.V2.Managers.BoxResourceManager.GetNetCoreVersion() method:

```
private string GetNetCoreVersion()
{
    var assembly = typeof(System.Runtime.GCSettings).GetTypeInfo().Assembly;
    
    if (assembly?.CodeBase != null) // -- exception is thrown here
    {
        var assemblyPath = assembly.CodeBase.Split(new[] { '/', '\\' }, StringSplitOptions.RemoveEmptyEntries);
        var netCoreAppIndex = Array.IndexOf(assemblyPath, "Microsoft.NETCore.App");
        return netCoreAppIndex > 0 && netCoreAppIndex < assemblyPath.Length - 2 ?
            assemblyPath[netCoreAppIndex + 1] :
            null;
    }

#if NETSTANDARD2_0
    var frameworkVersion = System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription;
    return Regex.Match(frameworkVersion, @"\d+(\.\d+)+").Value;
#else
    return null;
#endif
}
```

### Error Message, Including Stack Trace
System.NotSupportedException: CodeBase is not supported on assemblies loaded from a single-file bundle.
   at System.Reflection.RuntimeAssembly.get_CodeBase()
   at Box.V2.Managers.BoxResourceManager.GetNetCoreVersion()
   at Box.V2.Managers.BoxResourceManager.GetEnvNameAndVersion()
   at Box.V2.Managers.BoxResourceManager.GetBoxUAHeader()
   at Box.V2.Managers.BoxResourceManager.AddDefaultHeaders(IBoxRequest request)
   at Box.V2.Managers.BoxResourceManager.ToResponseAsync[T](IBoxRequest request, Boolean queueRequest, IBoxConverter converter)
   at Box.V2.Managers.BoxUsersManager.GetCurrentUserInformationAsync(IEnumerable`1 fields)

### Versions Used
.Net SDK: 7.0.302
Windows:  10.0.19045

Please let me know if I need to provide any additional information, I'd be happy to help.

**Comment by mwwoda** [comment]
I'll publish a new package when the fix is merged.
Alternatively, you can try setting the `IncludeAllContentForSelfExtract` property in the .csproj file to true as a temporary workaround.

**Comment by mwwoda** [comment]
Fix should be introduced in [5.2.2](https://www.nuget.org/packages/Box.V2.Core/5.2.2). Let me know if you encounter any problems.

**Comment by enotesynctest2** [comment]
@mwwoda I tried it and it works as expected! Thanks so much for the quick fix.
